test/errorHandling.js

```diff
@@ -27,6 +37,8 @@ describe('#errorHandling', () => {
           assert.ok(err);
           if (! seenError) {
             seenError = true;
+            // do not wait on closing the broken statsd connection
+            statsd = null;
```

Comment:
Some changes I had to make in here for uds, but they all work in other protocols and felt reasonable (and minor)